.pre-commit-config.yaml

```diff
@@ -118,11 +118,11 @@ repos:
         pass_filenames: false
       - id: eslint
         name: eslint
-        files: \.(js|ts|vue)$
+        files: (frontend|automations).*?\.(js|ts|vue|json5|json)$
```

Comment:
Do we need to remove the (frontend|automations) from here?

Reply:
The `files` setting is used to tell pre-commit when to run ESLint. We only want it to run when there are changes to files that actually fall within ESLint's purview. But this change to add frontend and automations will exclude the repo-root configuration files (prettier, ESLint, and pnpmfile configs).

However, adding `json` and `json5` will start it to match files we don't want the ESLint hook to run for. Off the top of my head I know there are JSON fixtures in both the catalog and API unit tests. So we do need to restrict it at least somewhat more than just any JS file.

Perhaps something along these lines would work: https://regexr.com/7f1i4

It's annoyingly complicated to use a regex for this rather than an array of glob patterns, which would be a little cleaner:

```
[
  "{prettier.config,.eslintrc,pnpmfile}.{cjs,js}",
  "{frontend,automations}/**/*.{js,ts,vue,json,json5}",
]
```

They're effectively the same settings, but the glob is way easier to understand, imo :confused: I may also have overcomplicated that regex. It's hard to figure out how best to express run on any JS/TS/Vue file change in any directory and any JSON file changed only in these directories. Maybe you have some better ideas than me :crossed_fingers:

Reply:
ESLint FlatConfig will make this easier only because it will allow us to remove the duplicated file targeting from the ESLint command. Right now the base config for ESLint will apply to _every file it finds_ so we need to tell it only to look in valid places. This forces us to duplicate this configuration when we also want to tell pre-commit to only run ESLint when those particular files are changing.